Fix optc-gen.awk membership tests for gawk 5.4.1 compatibility.

This is the textbook example of incorrectly checking an element's existence in
an array.
https://www.gnu.org/software/gawk/manual/html_node/Reference-to-Elements.html

Not yet fixed upstream in gcc.

diff --git a/gcc/optc-gen.awk b/gcc/optc-gen.awk
index fb4e5c1..1a5f00c 100644
--- a/gcc/optc-gen.awk
+++ b/gcc/optc-gen.awk
@@ -86,7 +86,7 @@ for (i = 0; i < n_opts; i++) {
                     }
                     condition = "opts->x_" opt_var_name_1 " && opts->x_" opt_var_name_2;
                 }
-                if (enables[enabledby_name] == "") {
+                if (!(enabledby_name in enables)) {
                     enabledby[n_enabledby] = enabledby_name;
                     n_enabledby++;
                 }
@@ -141,7 +141,7 @@ for (i = 0; i < n_opts; i++) {

 		lang_name = lang_sanitized_name(lang_name);

-		if (enables[lang_name,enabledby_array[k]] == "") {
+		if (!((lang_name,enabledby_array[k]) in enables)) {
 		    enabledby[lang_name,n_enabledby_lang[lang_index]] = enabledby_array[k];
 		    n_enabledby_lang[lang_index]++;
 		}
